### ISSUE TITLE
Stitching does not escape backslashes consistently

### ISSUE THREAD

**Issue by pellekrogholt** [ghostdogpr/caliban#1794]
Problem: 

Stitching graphs with Json as inputs that contain `\n`, `\r`, `\”` and so forth is not correctly escaped in `RemoteQuery.QueryRenderer.render(_)` at least some invalid json is produced and breaks stitching.

This stitching code is fragile:

```scala
  case class StringValue(value: String)   extends Value {
    override def toString: String = s""""${value.replace("\"", "\\\"").replace("\n", "\\n")}""""
  }
```

Reproduced here: https://scastie.scala-lang.org/wJwNYBRVR1aYJPsaDU0mvQ

**Comment by ghostdogpr** [comment]
I believe it was fixed in https://github.com/ghostdogpr/caliban/pull/1835

**Comment by emilgodsk** [comment]
Hi @ghostdogpr - I do not believe it was fixed in #1835 

See the updated Scastie using the new render https://scastie.scala-lang.org/C1nQo987RU2Sm8NLrisWbA as well as my fork, adding some tests: https://github.com/ghostdogpr/caliban/compare/series/2.x...emilgodsk:caliban:series/2.x

Here, the following tests are failing:
```
test("correctly renders a query for a field of json escaped characters")
test("correctly renders a query for a field of json escaped escaped quote")
test("correctly renders a query for a field of json escaped newline")
test("correctly renders a query for a field of json escaped carriage return")
test("correctly renders a query for a field of json escaped tab")
```

The test `test("correctly renders a query for a field of json escaped characters")` basically tests all the escaped characters at once. Here it is worth noting, that I added circe as a test dependency, to validate that the JSON we expect is valid, however, the actual output is invalid.

**Comment by ghostdogpr** [comment]
@paulpdaniels if you have a chance to take a look
